test/functional/news_controller_test.rb

```diff
@@ -5,6 +5,7 @@ class NewsControllerTest < ActionController::TestCase
     @rubygem1 = create(:rubygem, downloads: 10)
     @rubygem2 = create(:rubygem, downloads: 20)
     @rubygem3 = create(:rubygem, downloads: 30)
+    10.times { |i| create(:version, rubygem: @rubygem2, created_at: 5.days.ago, platform: "platform#{i}") }
```

Comment:
This definitely slows these tests down quite a bit. Any thoughts on setting up this state only once and sharing it among the tests in `NewsControllerTest`?

Reply:
Why 10, isn't 2 enough?

Reply:
By only creating 2 records here (and thus 3 `versions` records corresponding to `@rubygem2`), that would limit the total number of `versions` records to 5. On master, the SQL query generated by `news#show` limits the results to 10 records, meaning that there would still be room in the result for the 3 duplicate records corresponding to `@rubygem2` and the single record corresponding to `@rubygem3`. This would get returned by Active Record as `[#<@rubygem2 instance>, #<@rubygem3 instance>`, which doesn't catch the bug.

Because they are the most recent and on different platforms, the 10 records corresponding to `@rubygem2` essentially occupy all 10 spots available in the SQL query result. Active Record then interprets this result as just `[#<@rubygem2>]`.

However, now that you've brought it up, I think it would be better to replace 10 with `Gemcutter::NEWS_PER_PAGE` in case it were to change.

Reply:
Tests with your changes will catch this bug, but lets a add separate test just for this bug. You can move this `10.times` in a new `context "with multiple latest versions on different platfroms" should "display Gemcutter::NEWS_PER_PAGE entries"`. 
Definitely use defined constants instead of numbers, `@news = Rubygem.news(7.days)` in unit test should be updated as well.